examples/hosts/app-integration/schema-upgrade/src/view/debugView.tsx

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React, { useEffect, useState } from "react";
+
+import { IMigratableModel, MigrationState } from "../migrationInterfaces";
+
+export interface IDebugViewProps {
+    model: IMigratableModel;
+}
+
+export const DebugView: React.FC<IDebugViewProps> = (props: IDebugViewProps) => {
+    const {
+        model,
+    } = props;
+
+    return (
+        <div>
+            <h2 style={{ textDecoration: "underline" }}>Debug info</h2>
+            <MigrationStatusView model={ model } />
+            <ControlsView proposeVersion={ model.proposeVersion } />
+        </div>
+    );
+};
+
+interface IMigrationStatusViewProps {
+    model: IMigratableModel;
+}
+
+const MigrationStatusView: React.FC<IMigrationStatusViewProps> = (props: IMigrationStatusViewProps) => {
+    const { model } = props;
+
+    const [migrationState, setMigrationState] = useState<MigrationState>(model.getMigrationState());
+
+    useEffect(() => {
+        const migrationStateChangedHandler = () => {
+            setMigrationState(model.getMigrationState());
+        };
+        model.on("migrating", migrationStateChangedHandler);
+        model.on("migrated", migrationStateChangedHandler);
+        migrationStateChangedHandler();
+        return () => {
+            model.off("migrating", migrationStateChangedHandler);
+            model.off("migrated", migrationStateChangedHandler);
+        };
+    }, [model]);
+
+    return (
+        <div style={{ margin: "10px 0" }}>
+            <div>
+                Using model: { model.version }
+            </div>
+            <div>
+                Status:
+                { migrationState === MigrationState.collaborating && " Normal collaboration" }
+                { migrationState === MigrationState.migrating && " Migration in progress" }
+                { migrationState === MigrationState.migrated && " Migration complete" }
+            </div>
+            <div>
+                {
+                    model.acceptedVersion === undefined
+                        ? "No migration proposed yet"
+                        : `Proposed version to migrate to: ${model.acceptedVersion}`
+                }
+            </div>
+            <div>
+                {
+                    model.newContainerId === undefined
+                        ? "No migrated container yet"
+                        : `Migrated to new container at ${model.newContainerId}`
```

Comment:
Might be nice to turn this into a clickable URL that opens the new container in a new tab. Obviously not necessary since this is just a demo though

Reply:
Added in latest